### DISCUSSION TITLE
WebRTC doesn't work on raspberry pi4B.

### DISCUSSION THREAD

**Discussion by bakusui** [Igalia/cog#520] [General]
https://awslabs.github.io/amazon-kinesis-video-streams-webrtc-sdk-js/examples/index.html
Windows PC master -> rapsberry pi viewer.
result confirmed by.

```
https://awslabs.github.io/amazon-kinesis-video-streams-webrtc-sdk-js/examples/app.js:34:27: CONSOLE LOG [VIEWER] Stopping viewer connection
https://awslabs.github.io/amazon-kinesis-video-streams-webrtc-sdk-js/examples/app.js:34:27: CONSOLE LOG [VIEWER] Channel ARN: ******** ****************(edited)
https://awslabs.github.io/amazon-kinesis-video-streams-webrtc-sdk-js/examples/app.js:34:27: CONSOLE LOG [VIEWER] Endpoints: [object Object]
https://awslabs.github.io/amazon-kinesis-video-streams-webrtc-sdk-js/examples/app.js:34:27: CONSOLE LOG [VIEWER] ICE servers: [object Object],[object Object ],[object Object]
https://awslabs.github.io/amazon-kinesis-video-streams-webrtc-sdk-js/examples/app.js:22:29: CONSOLE ERROR ReferenceError: Can't find variable: RTCPeerConnection
```

https://github.com/Igalia/meta-webkit/wiki/RPi
Built with reference to.
WPE ver : 2.36+git%, 2.38.2+git%
Add the following to local.conf.
PACKAGECONFIG: append_pn-wpewebkit = " mediasource video webrtc mediastream "
PREFERRED_VERSION_wpewebkit = "2.36.0+git%"

I was able to connect with the i.MX8M mini(WPE ver:2.36+git%).

**Comment by philn** [comment]
You need to enable it at runtime too, there's a webrtc websetting and also one for mediastream.
Regarding the current status(-quo), see https://wpewebkit.org/about/faq.html#what’s-the-status-regarding-webrtc?

**Comment by bakusui** [comment]
What should I do specifically?
Now I am doing:

> export QT_QPA_PLATFORM=wayland
export XDG_RUNTIME_DIR=/run/user/0
COG_PLATFORM_FDO_VIEW_FULLSCREEN=1 cog -P wl  --ignore-tls-errors --set-permissions all --enable-write-console-messages-to-stdout true  --allow-file-access-from-file-urls true --allow-universal-access-from-file-urls true --allow-top-navigation-to-data-urls true https://awslabs.github.io/amazon-kinesis-video-streams-webrtc-sdk-js/examples/index.html

**Comment by philn** [comment]
If you check the `--websettings-help` output you should see at least a `mediastream` option listed... Enable with `--enable-mediastream=1`.

**Reply by bakusui** [reply]
I'm not doing well.
Could you teach me how to create a "yocto environment" in which WebRTC runs on a Raspberry Pi 4B?

**Comment by bakusui** [comment]
I understand that setting as default ON.
It was the same error even if I added option specification.
I added other options, but error.

COG_PLATFORM_FDO_VIEW_FULLSCREEN=1 cog -P wl --enable-media-stream true --enable-media true --enable-mediasource true --enable-media-capabilities true --ignore-tls-errors --set-permissions all --enable-write-console-messages-to-stdout true  --allow-file-access-from-file-urls true --allow-universal-access-from-file-urls true --allow-top-navigation-to-data-urls true

**Comment by bakusui** [comment]
Sorry the below didn't work.(and " 2.38.2+git%")
I will check with "2.38+https%" from now on.

PREFERRED_VERSION_wpewebkit = "2.36.0+git%"

**Comment by bakusui** [comment]
2.38+https% : build error("ValueError: No archives for setuptools-56 found")...

Other source checked.
"kirkstone" base tree + meta-webkit(honistor) : 2.36+https% build ok.
-> result : same error("Can't find variable: RTCPeerConnection")

**Reply by philn** [reply]
Please share the full list of WPE build flags.

**Reply by bakusui** [reply]
Can the value of log.do_configure be what you want?

> --  ENABLE_ACCESSIBILITY .......................... OFF
> --  ENABLE_BUBBLEWRAP_SANDBOX                       OFF
> --  ENABLE_ENCRYPTED_MEDIA ........................ OFF
> --  ENABLE_GTKDOC                                   OFF
> --  ENABLE_JOURNALD_LOG ........................... OFF
> --  ENABLE_MEDIA_SOURCE                             ON
> --  ENABLE_VIDEO .................................. ON
> --  ENABLE_WEBDRIVER                                ON
> --  ENABLE_WEB_AUDIO .............................. ON
> --  ENABLE_WEB_CRYPTO                               ON
> --  ENABLE_WPE_QT_API ............................. OFF
> --  ENABLE_XSLT                                     ON
> --  USE_ANGLE_WEBGL ............................... OFF
> --  USE_AVIF                                        OFF
> --  USE_JPEGXL .................................... OFF
> --  USE_LCMS                                        OFF
> --  USE_OPENJPEG .................................. ON
> --  USE_SOUP2                                       ON
> --  USE_WOFF2 ..................................... ON

**Reply by philn** [reply]
That's not the full list. Look for cmakeconfig.h and attach it here.....

**Reply by bakusui** [reply]
Thanks,this file?
[cmakeconfig.zip](https://github.com/Igalia/cog/files/10206509/cmakeconfig.zip)

**Reply by philn** [reply]
`ENABLE_MEDIA_STREAM` is 0 instead of 1. Same for `ENABLE_WEB_RTC`...

**Reply by bakusui** [reply]
Thank you, I made a mistake.
PACKAGECONFIG: append_pn-wpewebkit -> PACKAGECONFIG: append:pn-wpewebkit

**Reply by bakusui** [reply]
The original error is gone. But Gstreamer doesn't seem to work. I'll look into it some more.
